projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/datajobs/SwaggerConfiguration.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.datajobs;
+
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.servers.Server;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+
+import javax.annotation.PostConstruct;
+import java.util.List;
+
+/**
+ * This class exists so that when running quickstart vdk the swagger ui has the correct base url.
```

Comment:
This is not a problem unique to quickstart vdk. But to any deployment that is going to be similar to quickstart vdk. I would generalize the description since and use quickstart vdk as an example for one such case.

Reply:
Well it is unique to kind. 
When running behind a reverse proxy/ingress it can actually look at the forward-x headers. 
Even if service is running behind multiple layers of ingress it will most likely get the url correct. 


I don't see other setting it up in kind except when using quckstart vdk.

Reply:
WDYT?

Reply:
My advice is not to assume how people would set it up. You can never know. Likelyhood is a lot of people would do something we do not expect. 

So my advice is to explain (to the best of your knowledge) under what setup would this need to be set. If there's no accurate way to describe it (there's a lot of uncertainty) . Or you can just say - Used if swagger URL is not autodetected correctly for any reason just set it using this variable. 
 
Another setup that is not autodetected correctly is https://iaclqhm5xk.execute-api.us-west-1.amazonaws.com/data-jobs/swagger-ui/index.html#/ 

You would notice the Swagger URL is the non secure (http) one instead of the API gateway provided URL. Things work because the non-secure one is also public at the moment.

Reply:
Please resolve if you think it looks better now.